deployment/aws-python-storage-stack-automation/storage_stack_lifecycle.yaml

```diff
@@ -352,6 +352,244 @@ Resources:
+              ext_id = ext["externalID"]
+              print("CloudOne ExtID: " + ext_id)
+              
+              # gather aws account ID
+              account_id = event["account"]
+              print("AWS account ID: " + account_id)
+
+              #gather stack id
+              has_next_value = False
+              next = ""
+              #default limit is 25
+              q_params = {
+                "provider": "aws",
+                "status": "ok",
+                "type": "storage"
+              }
+              id_call = http.request('GET', stacks_url, q_params, headers = {'Content-Type': 'application/json', 'api-secret-key': ws_key, 'Api-Version': 'v1'})
+              try:
+                  id_resp = json.loads(id_call.data.decode('utf-8'))
+              except json.decoder.JSONDecodeError:
+                  time.sleep(1)
+                  id_resp = json.loads(id_call.data.decode('utf-8'))
+              for data in id_resp['stacks']:
+                  if 'storage' in data and data['storage'] is not None:
+                      if bucket_name == data['storage']:
+                          stack_name = data['name']
+                          stack_id = data['stackID']
+                          print(stack_name)
+                          print(stack_id)
+                          stack_removal(ws_key, stack_id,stack_name)
+                      if "next" in id_resp:
+                        has_next_value = True
+                        next = id_resp['next']
+                      while has_next_value:
+                        #next call
+                        q_params = {
+                          "provider": "aws",
+                          "status": "ok",
+                          "type": "storage",
+                          "cursor": next
+                        }
+                        id_call = http.request('GET', stacks_url, q_params, headers = {'Content-Type': 'application/json', 'api-secret-key': ws_key, 'Api-Version': 'v1'})
+                        try:
+                          id_resp = json.loads(id_call.data.decode('utf-8'))
+                        except:
+                          time.sleep(1)
+                          id_resp = json.loads(id_call.data.decode('utf-8'))
+                        for data in id_resp['stacks']:
+                          if 'storage' in data and data['storage'] is not None:
+                            if bucket_name == data['storage']:
+                              stack_name = data['name']
+                              stack_id = data['stackID']
+                              print(stack_name)
+                              print(stack_id)
+                              stack_removal(ws_key, stack_id, stack_name)
+                              has_next_value = False
+                        if "next" in id_resp:
+                          next = id_resp['next']
+                        else:
+                          has_next_value = False
```

Comment:
I think you didn't get what I meant 100%. I meant something like
```suggestion
              if "next" in id_resp:
                has_next_value = True
                next = id_resp['next']
              while has_next_value:
                #next call
                q_params = {
                  "provider": "aws",
                  "status": "ok",
                  "type": "storage",
                  "cursor": next
                }
                id_call = http.request('GET', stacks_url, q_params, headers = {'Content-Type': 'application/json', 'api-secret-key': ws_key, 'Api-Version': 'v1'})
                try:
                  id_resp = json.loads(id_call.data.decode('utf-8'))
                except:
                  time.sleep(1)
                  id_resp = json.loads(id_call.data.decode('utf-8'))
                for data in id_resp['stacks']:
                  if 'storage' in data and data['storage'] is not None:
                    if bucket_name == data['storage']:
                      stack_name = data['name']
                      stack_id = data['stackID']
                      print(stack_name)
                      print(stack_id)
                      stack_removal(ws_key, stack_id, stack_name)
                      has_next_value = False
                if "next" in id_resp:
                  next = id_resp['next']
                else:
                  has_next_value = False
```

Reply:
Also, please don't mix use 2-space and 4-space indents in python.

Reply:
I changed the indentation to match across.

Reply:
You still didn't get what I meant. The `if "next" in id_resp:` block should be outside of the loop at line#515

Reply:
Sorry. I was not entirely sure what you meant. I have pulled the if statement out of the for loop at 515.